### ISSUE TITLE
Future development

### ISSUE THREAD

**Issue by kristoforsalmin** [kraftvaerk/generator-rammevaerk#54]
This is a tracking issue for our future work on rammevaerk. After a talk with @mi2oon we came up with a following list of improvements, please submit yours in the comments below.

**Build system**

- [ ] Replace gulp with webpack
- [ ] Support multiple configs with `webpack-merge`
- [ ] Support multiple Browserslist environments (modern and legacy browsers). Look into reusing something like [`browserslist-useragent-regexp`](https://github.com/browserslist/browserslist-useragent-regexp) on the backend
- [ ] Generate `assets-manifest.json`
- [ ] Support `.env` files via [`dotenv`](https://github.com/motdotla/dotenv) or something similar
- [ ] Figure out best way to serve mockups (via CLI, `webpack-dev-server` or something else)
- [ ] Figure out best way to lint scripts and styles (via CLI, during the build, pre-commit hook, etc.)
- [ ] Check state of sourcemaps for CSS
- Maybe include Vue.js setup by default

**Generator**

- [ ] Adjust project structure for webpack (folders, config names, etc.)
- [ ] Look into [`create-social-network`](https://github.com/udilia/create-social-network) and [`create-react-app`](https://github.com/facebook/create-react-app) to learn more about scaffolding
- Maybe remove Yeoman

**Comment by mi2oon** [comment]
Also lets consider to add `prettierrc` and `jsconfig`

**Comment by kristoforsalmin** [comment]
@mi2oon yeah, Prettier's definitively a nice idea (added to the top) 👍 Regarding `jsconfig.json`, do you wish to use it for code checking or just aliasing?

**Comment by mi2oon** [comment]
@kristoforsalmin I was thinking to use it for as simple as excluding node_modules folder and also aliasing.

**Comment by mi2oon** [comment]
@kristoforsalmin I've created a [new branch](https://github.com/kraftvaerk/generator-rammevaerk/tree/next) for this. Also I've started looking at the generator side of the task. I commit my latest experiments soon 🔬:boom: 😉

**Comment by mi2oon** [comment]
![Image of Rammevaerk](https://public.boxcloud.com/api/2.0/internal_files/539387499829/versions/571596074629/representations/jpg_paged_2048x2048/content/1.jpg?access_token=1!7RsQM2u09fpzOW0w0UP2oI7_XIwYEC7C8qilCCZFOhewdaNfQTrkef7DZpPi0ECF4jXm0MKo0jgcZOE9vLIb69eey5r69CvmHZrpHtTMGtAD4l60N3lc-oBPM3jZVEc_tElDVLKSfu6TmyHI7LHhulLmqeqOVODFcMJB3MeVEmFahER4MN4EMxcV64zj2ozB8jr6qG7aQkHkzAdvuEiTAWDjpyhTtOBAObUi4wl1pKMfIaqckfjSe1pxKtZjKSBqXcL6Teofib6s7c1tgFGtr8xgP2lXQcoPrEei34DxmPnsQBYfI2TuZuUjGG8jI-tVLSq2YXe0T5CsOEzPjzUIQUxP1VtQxy5RX4aV98O0aiypkmpLBu39Uv-0TyiAdLwsCREeHYdL75J7CwASo2bJs0iTtLxTYopKE3bkOpO01z6mOOk1IJ6D1-npxMCrg30hrjKclxGRD9pfgRDwTzB0ZU696m-5UBROXnXjisRtjpIywgFQa_YPMlBMV8M9oLizTI8BbQD4W08w--h9XKaSYl4aOT5E_EZDz1Wch9-zptFYjImKEZijP4u7e8ZfEp0.&box_client_name=box-content-preview&box_client_version=2.22.0)

Sneak peak.. missing to handle tests but otherwise, I think I am almost done ;)

**Comment by kristoforsalmin** [comment]
@mi2oon cool! Is that an official Rammevaerk mascot? 😁 I created a project for this release. I feel like issues aren't convenient for tracking progress and so on, anyways have a look https://github.com/kraftvaerk/generator-rammevaerk/projects/2.

**Comment by kristoforsalmin** [comment]
@mi2oon that looks awesome (now can see the screenshot) 👍
